### ISSUE TITLE
SIGSEGV occurs in RaftConsensus::DeduplicateLeaderRequestUnlocked

### ISSUE THREAD

**Issue by ajcaldera1** [yugabyte/yugabyte-db#3529]
Found during CDC stress testing on 2.0.11.0-b23:
```
#0  0x00007f4be5c29637 in id (this=0x4eb89a80) at ../../yb/consensus/consensus.pb.h:5270
#1  yb::consensus::RaftConsensus::DeduplicateLeaderRequestUnlocked (this=this@entry=0x13fdc5b0, rpc_req=rpc_req@entry=0xb6acfd0, deduplicated_req=deduplicated_req@entry=0x7f4b6ad79d00)
    at ../../../../../src/yb/consensus/raft_consensus.cc:1309
#2  0x00007f4be5c383ce in yb::consensus::RaftConsensus::CheckLeaderRequestUnlocked (this=this@entry=0x13fdc5b0, request=request@entry=0xb6acfd0, response=response@entry=0x1c40ec10, deduped_req=deduped_req@entry=0x7f4b6ad79d00)
    at ../../../../../src/yb/consensus/raft_consensus.cc:1437
#3  0x00007f4be5c388dd in yb::consensus::RaftConsensus::UpdateReplica (this=this@entry=0x13fdc5b0, request=request@entry=0xb6acfd0, response=response@entry=0x1c40ec10) at ../../../../../src/yb/consensus/raft_consensus.cc:1601
#4  0x00007f4be5c39ab3 in yb::consensus::RaftConsensus::Update (this=this@entry=0x13fdc5b0, request=request@entry=0xb6acfd0, response=response@entry=0x1c40ec10, deadline=...)
    at ../../../../../src/yb/consensus/raft_consensus.cc:1204
#5  0x00007f4be67f2cf1 in yb::tserver::ConsensusServiceImpl::UpdateConsensus (this=this@entry=0x2b67800, req=req@entry=0xb6acfd0, resp=resp@entry=0x1c40ec10, context=...) at ../../../../../src/yb/tserver/tablet_service.cc:1666
#6  0x00007f4be43ee6aa in yb::consensus::ConsensusServiceIf::Handle (this=0x2b67800, call=...) at consensus.service.cc:100
#7  0x00007f4be0684c19 in yb::rpc::ServicePoolImpl::Handle (this=0x2defb00, incoming=...) at ../../../../../src/yb/rpc/service_pool.cc:262
#8  0x00007f4be06288f4 in yb::rpc::InboundCall::InboundCallTask::Run (this=<optimized out>) at ../../../../../src/yb/rpc/inbound_call.cc:212
#9  0x00007f4be0690868 in yb::rpc::(anonymous namespace)::Worker::Execute (this=<optimized out>) at ../../../../../src/yb/rpc/thread_pool.cc:99
#10 0x00007f4bdec59e0f in operator() (this=0x5e2d378) at /home/yugabyte/yb-software/yugabyte-2.0.11.0-b23-centos-x86_64/linuxbrew-xxxxxxxxxxxx/Cellar/gcc/5.5.0_4/include/c++/5.5.0/functional:2267
#11 yb::Thread::SuperviseThread (arg=0x5e2d320) at ../../../../../src/yb/util/thread.cc:739
#12 0x00007f4bd9718694 in start_thread (arg=0x7f4b6ad7b700) at pthread_create.c:333
#13 0x00007f4bd8e5541d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

**Comment by tylarb** [comment]
Believe I've found this issue on 2.2.7 as well. 

Core dump: 

```
Core was generated by `/home/yugabyte/tserver/bin/yb-tserver --flagfile /home/yugabyte/tserver/conf/se'.
Program terminated with signal 11, Segmentation fault.
#0  0x00007fb670810dd7 in id (this=0x342706120) at src/yb/consensus/consensus.pb.h:5453
5453    src/yb/consensus/consensus.pb.h: No such file or directory.
(gdb) bt
#0  0x00007fb670810dd7 in id (this=0x342706120) at src/yb/consensus/consensus.pb.h:5453
#1  yb::consensus::RaftConsensus::DeduplicateLeaderRequestUnlocked (this=this@entry=0x2266b50, 
    rpc_req=rpc_req@entry=0x26bb749a0, deduplicated_req=deduplicated_req@entry=0x7fac085cccb0)
    at ../../src/yb/consensus/raft_consensus.cc:1399
#2  0x00007fb67081ff1e in yb::consensus::RaftConsensus::CheckLeaderRequestUnlocked (this=this@entry=0x2266b50, 
    request=request@entry=0x26bb749a0, response=response@entry=0x189d14e50, deduped_req=deduped_req@entry=0x7fac085cccb0)
    at ../../src/yb/consensus/raft_consensus.cc:1527
#3  0x00007fb67082043b in yb::consensus::RaftConsensus::UpdateReplica (this=this@entry=0x2266b50, 
    request=request@entry=0x26bb749a0, response=response@entry=0x189d14e50) at ../../src/yb/consensus/raft_consensus.cc:1691
#4  0x00007fb6708215d3 in yb::consensus::RaftConsensus::Update (this=this@entry=0x2266b50, 
    request=request@entry=0x26bb749a0, response=response@entry=0x189d14e50, deadline=...)
    at ../../src/yb/consensus/raft_consensus.cc:1294
#5  0x00007fb67149fd51 in yb::tserver::ConsensusServiceImpl::UpdateConsensus (this=this@entry=0x15fc720, 
    req=req@entry=0x26bb749a0, resp=resp@entry=0x189d14e50, context=...) at ../../src/yb/tserver/tablet_service.cc:2116
#6  0x00007fb66dd54d4a in yb::consensus::ConsensusServiceIf::Handle (this=0x15fc720, call=...)
    at src/yb/consensus/consensus.service.cc:100
#7  0x00007fb669ac5ed9 in yb::rpc::ServicePoolImpl::Handle (this=0x19a5200, incoming=...)
    at ../../src/yb/rpc/service_pool.cc:262
#8  0x00007fb669a6b6b4 in yb::rpc::InboundCall::InboundCallTask::Run (this=<optimized out>)
    at ../../src/yb/rpc/inbound_call.cc:212
#9  0x00007fb669ad1c58 in yb::rpc::(anonymous namespace)::Worker::Execute (this=<optimized out>)
    at ../../src/yb/rpc/thread_pool.cc:100
#10 0x00007fb668301f9f in operator() (this=0xc8f45f8)
    at /home/yugabyte/yb-software/yugabyte-2.2.7.0-b5-centos-x86_64/linuxbrew-xxxxxxxxxxxxxx/Cellar/gcc/5.5.0_4/include/c++/5.5.0/functional:2267
#11 yb::Thread::SuperviseThread (arg=0xc8f45a0) at ../../src/yb/util/thread.cc:759
#12 0x00007fb662b1e694 in start_thread (arg=0x7fac085ce700) at pthread_create.c:333
#13 0x00007fb66225b41d in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:109
```

**Comment by bmatican** [comment]
Seems this was fixed in c9b4a3ce64d437b248baa6a262af07bdce22bc2e, but this is only in 2.3.0, closing for now, can discuss internally if to port to 2.2
